lib/rules/space-infix-ops.js

```diff
@@ -75,26 +116,31 @@ module.exports = {
          * @private
          */
         function report(mainNode, culpritToken) {
+            const operator = culpritToken.value;
+            const must = spacesYes[operator] ? "must" : "must not";
+
             context.report({
                 node: mainNode,
                 loc: culpritToken.loc.start,
-                message: "Infix operators must be spaced.",
+                data: { must, operator },
+                message: "Operator `{{operator}}` {{must}} be surrounded by spaces.",
                 fix(fixer) {
+                    const shouldHaveSpaces = spacesYes[operator];
                     const previousToken = sourceCode.getTokenBefore(culpritToken);
-                    const afterToken = sourceCode.getTokenAfter(culpritToken);
-                    let fixString = "";
+                    const nextToken = sourceCode.getTokenAfter(culpritToken);
 
-                    if (culpritToken.range[0] - previousToken.range[1] === 0) {
-                        fixString = " ";
-                    }
+                    let fixString = sourceCode.getText(previousToken);
 
+                    if (shouldHaveSpaces) {
+                        fixString += " ";
+                    }
                     fixString += culpritToken.value;
-
-                    if (afterToken.range[0] - culpritToken.range[1] === 0) {
+                    if (shouldHaveSpaces) {
                         fixString += " ";
                     }
+                    fixString += sourceCode.getText(nextToken);
 
-                    return fixer.replaceText(culpritToken, fixString);
+                    return fixer.replaceText(mainNode, fixString);
```

Comment:
Also, we should make sure that the fixer doesn't cause syntax errors:

```js
/* eslint space-infix-ops: [error, {"all": "never"}] */

foo in bar;
foo instanceof bar;
foo + +bar;
foo - -bar;

// these should get fixed to

foo in bar;
foo instanceof bar;
foo+ +bar;
foo- -bar;
```

Removing *all* of the spaces in those examples would result in a SyntaxError.

To account for this, I think the fixer needs to have logic similar to the [no-extra-parens](https://github.com/eslint/eslint/blob/b7432bdea386041012052621ab66758163d58ef4/lib/rules/no-extra-parens.js#L278-L294) fixer to determine whether it can safely remove a space.